lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp

```diff
@@ -687,6 +693,64 @@ void InitLtcModuleBindings(py::module m) {
   m.def("_ltc_enable_thread_pool", []() {
     FLAGS_torch_lazy_use_thread_pool = true;
   });
+  /*
+   * Return tensor ids and tensors for DeviceData nodes.
+   */
+  m.def("_get_ltc_tensors_ts_device_data_node",
+        [](const std::vector<at::Tensor>& tensors) {
+          std::vector<Node*> roots;
+          for (auto& tensor : tensors) {
+            auto xtensor = TryGetLtcTensor(tensor);
+            roots.push_back(xtensor.GetIrValue().node.get());
+          }
+          auto post_order = Util::ComputePostOrder(roots);
+          std::vector<int64_t> tensor_ids;
+          std::vector<at::IValue> ivalues;
+          for (auto nodeptr : post_order) {
+            if (nodeptr->op() == *torch::lazy::ltc_device_data) {
+              const auto* device_data_node = torch::lazy::NodeCast<torch::lazy::DeviceData>(nodeptr, *torch::lazy::ltc_device_data);
```

Comment:
could you comment/explain what the assumption/relationship is between ivalues/tensor_ids?  (Say why this mapping is important)

Reply:
Agreed, this is a bit subtle. AFAIU the mapping is important because JIT graphs accept inputs in the exact same order since lowering is also done in the exact same order and JIT graphs is what actually execute the fx graphs with their real inputs.